src/lazy/encoder/binary/v1_1/container_writers.rs

```diff
@@ -57,20 +58,26 @@ impl<'value, 'top> BinaryContainerWriter_1_1<'value, 'top> {
         start_opcode: u8,
         allocator: &'top BumpAllocator,
         buffer: &'value mut BumpVec<'top, u8>,
+        inline_symbol_text: bool,
```

Comment:
🗺️ Most of this patch is just plumbing a new `inline_symbol_text` option through the different binary 1.1 container writers and associated machinery.

Reply:
Is it worth having a `write_options: WriteOptions` (or similar) field instead so that any new options/flags can reuse the same plumbing rather than needing to add a new field in all of these different places?

Reply:
Yeah, that is better. I've moved over to a consolidated `ValueWriterConfig` type in https://github.com/amazon-ion/ion-rust/pull/804/commits/ef7c84545db248ee42760aa185fc428e183da56a.

It adds separate settings for:
* Encoding symbol values
* Encoding annotations sequences
* Encoding struct field names
* Delimited vs LP containers